# 使用 Anaconda

[Anaconda](http://www.continuum.io/downloads)是一个很好用的Python IDE，它集成了很多科学计算需要使用的**python**第三方工具包。

## conda 的使用 

根据自己的操作系统安装好[Anaconda](http://www.continuum.io/downloads)后，在命令行下输入：

    conda list

可以看已经安装好的**python**第三方工具包，这里我们使用 `magic` 命令 `%%cmd` 在 `ipython cell` 中来执行这个命令：

In [1]:
!conda list

# packages in environment at C:\Users\ogs\Anaconda2:
#
_license                  1.1                      py27_1  
_nb_ext_conf              0.3.0                    py27_0  
alabaster                 0.7.9                    py27_0  
anaconda                  4.2.0               np111py27_0  
anaconda-clean            1.0.0                    py27_0  
anaconda-client           1.5.1                    py27_0  
anaconda-navigator        1.3.1                    py27_0  
argcomplete               1.0.0                    py27_1  
astroid                   1.4.7                    py27_0  
astropy                   1.2.1               np111py27_0  
babel                     2.3.4                    py27_0  
backports                 1.0                      py27_0  
backports_abc             0.4                      py27_0  
beautifulsoup4            4.5.1                    py27_0  
bitarray                  0.8.1                    py27_1  
blaze                     0.10.1             

第一次安装好 [Anaconda](http://www.continuum.io/downloads) 以后，可以在命令行输入以下命令使 [Anaconda](http://www.continuum.io/downloads) 保持最新：

    conda update conda
    conda update anaconda

conda 是一种很强大的工具，具体用法可以参照它的[文档](http://conda.pydata.org/docs/)。

也可以参考它的 [cheat sheet](http://conda.pydata.org/docs/_downloads/conda-cheatsheet.pdf) 来快速查看它的用法。

可以使用它来安装，更新，卸载第三方的 **python** 工具包：

    conda install <some package>
    conda update <some package>
    conda remove <some package>

在安装或更新时可以指定安装的版本号，例如需要使用 `numpy 1.8.1`：

    conda install numpy=1.8.1
    conda update numpy=1.8.1

查看 `conda` 的信息：

    conda info

In [2]:
!conda info

Current conda install:

               platform : win-64
          conda version : 4.2.9
       conda is private : False
      conda-env version : 4.2.9
    conda-build version : 2.0.2
         python version : 2.7.12.final.0
       requests version : 2.11.1
       root environment : C:\Users\ogs\Anaconda2  (writable)
    default environment : C:\Users\ogs\Anaconda2
       envs directories : C:\Users\ogs\Anaconda2\envs
          package cache : C:\Users\ogs\Anaconda2\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64/
                          https://repo.continuum.io/pkgs/free/noarch/
                          https://repo.continuum.io/pkgs/pro/win-64/
                          https://repo.continuum.io/pkgs/pro/noarch/
                          https://repo.continuum.io/pkgs/msys2/win-64/
                          https://repo.continuum.io/pkgs/msys2/noarch/
            config file : None
           offline mode : False



一个很棒的功能是 `conda` 可以产生一个自定义的环境，假设在安装的是 **Python 2.7** 的情况下，想使用 **Python 3.4**，只需要在命令行下使用 `conda` 产生一个新的环境：

    conda create -n py34 python=3.4

这里这个环境被命名为 `py34` ，可以根据喜好将 `py34` 改成其他的名字。

使用这个环境时，只需要命令行下输入：

``` python
activate py34 #(windows)
source activate py34 #(linux, mac)
```

此时，我们的 **Python** 版本便是 **`python 3.4`**了。

## spyder 编辑器

`Anaconda` 默认使用的编辑器是 `spyder`，可以在命令行下输入：

    spyder

来进入这个编辑器，具体使用方法不做介绍。